src/NHibernate/Util/ExpressionsHelper.cs

```diff
@@ -225,10 +213,24 @@ public static class ExpressionsHelper
 							out currentComponentType);
 						break;
 					case IAbstractComponentType componentType:
+						// Concatenate the component property path in order to be able to use EntityMetamodel.GetPropertyType to retrieve the type.
+						// As GetPropertyType supports only components, do not concatenate when dealing with collection composite elements or elements.
+						if (!currentType.IsAnyType)
```

Comment:
Thanks for cleaning the code. I've removed this condition as it is no longer needed because `AnyType` also implements `IAssociationType`, so it will be handled by the above `case`.